assignments/this.js

```diff
@@ -1,26 +1,65 @@
-/* The for principles of "this";
-* in your own words. explain the four principle for the "this" keyword below.
-*
-* 1. 
-* 2. 
-* 3. 
-* 4. 
-*
-* write out a code example of each explanation above
-*/
-
-// Principle 1
+//  The for principles of "this";
+// * in your own words. explain the four principle for the "this" keyword below.
+// *
+// * 1. Principle 1: Window/Global Object Binding -
+//   >  “this” value is in the global scope of the window/console.
 
-// code example for Window Binding
+// * 2. Principle 2: Implicit Binding -
+//   >  Calling a function preceded by a dot, the object before that dot will be "this".
+
+// * 3.Principle 3: New binding -
+//   > When using a constructor function, "this" is referring to the specific instance of said object that was created and will be returned by the same constructor function.
+
+// * 4. Principle 4: Explicit binding -
+//   > When the methods `call` or `apply` is used in JavaScript , "this" is explicitly defined.
+
+
+// * write out a code example of each explanation above
 
-// Principle 2
 
+//  Principle 1
+// code example for Window Binding
+ function sayMyAge(age) {
+    console.log(this);
+    return age;
+  }
+  sayMyAge(38);
+
+//  Principle 2
 // code example for Implicit Binding
+const sayNameFunc = obj => {
+    obj.sayName = () => {
```

Comment:
This is the idea, with implicit binding.

You have a function nested within a parent function. 

The problem here is that this function doesn't work as intended. Gotta be a working function. Check out this example of implicit binding from the codepen:
```
const unsullied = {
  name: "Feeble Whelp",
  age: 31,
  weapon: "spear",
  hair: "none",
  disposition: "serious",
  defend: function() {
    console.log(this);
    return `${this.name} defends the queen, with his ${this.weapon}`;
  }
}

console.log(unsullied.defend())
```
https://codepen.io/letanque/pen/dEJdLr